docs/SemanticARC.md

```diff
@@ -92,20 +101,28 @@ Once we are able to reason about RC Identity, the next step in implementing Sema
 
    b. If the copy_value instruction has the [take] flag associated with it, then a move is being performed and while a bit by bit copy of the value occurs, no retain_value is applied to it. The original SSA value as a result of this operation has an undefined bit value and in debugging situations could be given a malloc scribbled payload.
 
+[EE] Why would we need a copy_value [take]?
+
 2. strong_release, release_value will be replaced by a destroy_value instruction with the following semantics:
 
    a. By default a destroy_value will perform a release_value on its input value. After this point, the bit value of the SSA value is undefinevd and in debugging situations, the SSA value could be given a malloc scribbled payload.
    
    b. A destroy_value with the [noop] flag attached to it does not perform a release_value on its input value but /does/ scribble over the memory in debugging situations. *FIXME [noop] needs a better name*.
 
+[EE] Can you please explain for what purpose destroy_value [noop] is used?
+
 3. strong store/strong load operations should be provided as instructions. This allows for normal loads to be considered as not having any ARC significant operations and eliminates a hole in ARC where a pointer is partially initialized (i.e. it a value is loaded but it has not been retained. In the time period in between those two points the value is partially initialized allowing for optimizer bugs).
 
+[EE] Would a store_strong also include the release of the old value?
```

Comment:
It is exactly like store_weak. If you do not pass in the initialization flag it does do that.